components/Forms/EditableUrl.tsx

```diff
@@ -9,54 +8,65 @@ interface Props {
   title: string
   isEditable: boolean
   textStyle: ButtonTypes
-  inputStyle?: any
+  inputStyle?: StyleProp<TextStyle>
   multiline: boolean
   placeholder?: string
-  onChange?(value: string): void
-  navigation?: any
-  route?: any
+  onChange(value: string): void
 }
-class EditableUrlImpl extends JCComponent<Props> {
-  onChanged(val: any) {
-    if (this.props.onChange) this.props.onChange(val.target.value)
-  }
-  navigate(id: string) {
-    // window.location.href = id
-    window.open(this.props.value, "_blank")
+
+interface State {
+  url: string
+}
+
+export default class EditableUrlImpl extends JCComponent<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      url: props.value,
+    }
   }
-  render() {
-    if (this.props.isEditable)
+
+  render(): JSX.Element | null {
+    const {
+      onChange,
+      isEditable,
+      placeholder,
+      multiline,
+      inputStyle,
+      textStyle,
+      value,
+      title,
+    } = this.props
+    const { url } = this.state
+
+    if (isEditable) {
       return (
         <TextInput
-          onChange={(value) => {
-            this.onChanged(value)
+          onChange={(e) => {
+            this.setState({ url: e.nativeEvent.text })
           }}
-          placeholder={this.props.placeholder}
-          multiline={this.props.multiline}
-          onStartShouldSetResponder={() => true}
```

Comment:
These guys were for situations where it might be in a touchable opacity....not sure if this component has any of those situations...there may be a better way of doing this, but it was the only solution I could find at the time...